exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Queries.kt

```diff
@@ -158,16 +164,31 @@ fun Table.deleteAll(): Int =
  * Represents the SQL statement that deletes rows in a table and returns specified data from the deleted rows.
  *
  * @param returning Columns and expressions to include in the returned data. This defaults to all columns in the table.
- * @param where Condition that determines which rows to delete. If left as `null`, all rows in the table will be deleted.
+ * @param where Condition that determines which rows to delete.
  * @return A [ReturningStatement] that will be executed once iterated over, providing [ResultRow]s containing the specified
  * expressions mapped to their resulting data.
  * @sample org.jetbrains.exposed.sql.tests.shared.dml.ReturningTests.testDeleteReturning
  */
-fun <T : Table> T.deleteReturning(
+inline fun <T : Table> T.deleteReturning(
     returning: List<Expression<*>> = columns,
-    where: (SqlExpressionBuilder.() -> Op<Boolean>)? = null
+    where: SqlExpressionBuilder.() -> Op<Boolean>
+): ReturningStatement {
+    val delete = DeleteStatement(this, SqlExpressionBuilder.where(), false, null)
+    return ReturningStatement(this, returning, delete)
+}
+
+/**
+ * Represents the SQL statement that deletes all rows in a table and returns specified data from the deleted rows.
+ *
+ * @param returning Columns and expressions to include in the returned data. This defaults to all columns in the table.
+ * @return A [ReturningStatement] that will be executed once iterated over, providing [ResultRow]s containing the specified
+ * expressions mapped to their resulting data.
+ * @sample org.jetbrains.exposed.sql.tests.shared.dml.ReturningTests.testDeleteReturning
+ */
+fun <T : Table> T.deleteReturning(
+    returning: List<Expression<*>> = columns
 ): ReturningStatement {
-    val delete = DeleteStatement(this, where?.let { SqlExpressionBuilder.it() }, false, null)
+    val delete = DeleteStatement(this, null, false, null)
     return ReturningStatement(this, returning, delete)
 }
```

Comment:
I created some new overloads to avoid issues with nullable parameters + `inline`.
I would have liked to be able to call one from the other instead of duplicating logic, but I couldn't get that to work given the nature of `SqlExpressionBuilder.() -> Op<Boolean>`.

@e5l Please let me know if any of these don't make sense to split just for inline.

Reply:
look ok